### PR TITLE
add public flag for wafpolicy multi-tenant

### PR DESCRIPTION
**Pull request by zhang-shengping**

### CODE REVIEW

File: app/waf/src/controllers/wafpolicy.controller.ts [hunk 1]
```diff
@@ -89,9 +89,12 @@ export class WafpolicyController extends BaseController {
   ): Promise<CollectionResponse> {
     return new CollectionResponse(
       Wafpolicy,
-      await this.wafpolicyRepository.find(filter, {
-        tenantId: await this.tenantId,
-      }),
+      await this.wafpolicyRepository.find(
+        {
```
Comment (zongzw) [inline app/waf/src/controllers/wafpolicy.controller.ts]: Add its test case?

Comment (zhang-shengping) [reply app/waf/src/controllers/wafpolicy.controller.ts]: we need multiple tenants in context and change them dynamically.

File: app/waf/src/controllers/wafpolicy.controller.ts [hunk 2]
```diff
@@ -109,8 +112,13 @@ export class WafpolicyController extends BaseController {
   ): Promise<Response> {
     return new Response(
       Wafpolicy,
-      await this.wafpolicyRepository.findById(id, undefined, {
-        tenantId: await this.tenantId,
+      await this.wafpolicyRepository.findById(id, {
+        where: {
+          and: [
+            {or: [{tenantId: await this.tenantId}, {public: true}]},
```
Comment (zongzw) [inline app/waf/src/controllers/wafpolicy.controller.ts]: Just thinking that `id` can identify anything. So there may be no need to `and` such 2 conditions. What do you think?

File: app/waf/src/controllers/wafpolicy.controller.ts (hunk 1 above)
Comment (zongzw) [reply app/waf/src/controllers/wafpolicy.controller.ts]: Why for that?

The test should be 
1. create the wafpolicy data whose property `public === true` and `tenantId === t1`.
2. access the api with `tenantId === t2` and check the return value is not empty and tenantId === t1. 

Is it?

File: app/waf/src/controllers/wafpolicy.controller.ts (hunk 2 above)
Comment (zhang-shengping) [reply app/waf/src/controllers/wafpolicy.controller.ts]: I hope so, but it does not.

Comment (zongzw) [reply app/waf/src/controllers/wafpolicy.controller.ts]: why "does not" ?

Comment (zhang-shengping) [reply app/waf/src/controllers/wafpolicy.controller.ts]: it becomes somethig like "{{or: [{xxx: xxx}, {xxx:xxx}]}, {"and":[{xxx:xxx}]}}"

File: app/waf/test/acceptance/wafpolicy.controller.acceptance.ts [hunk 3]
```diff
@@ -104,6 +104,49 @@ describe('WafpolicyController', () => {
     expect(toJSON(wafpolicy)).to.containDeep(response.body.wafpolicies[0]);
   });
 
+  it(
```
Comment (zongzw) [inline app/waf/test/acceptance/wafpolicy.controller.acceptance.ts]: Great.

File: app/waf/src/controllers/wafpolicy.controller.ts (hunk 2 above)
Comment (zongzw) [reply app/waf/src/controllers/wafpolicy.controller.ts]: What I mean is if user passes an Id, we find the record with only 1 condition(id), no other condition.

Comment (zhang-shengping) [reply app/waf/src/controllers/wafpolicy.controller.ts]: 1. we want to get a public a wafpolicy whose tenantId is not the same as we used now.
2. if its without any other condition, we may get a wafpolicy which belongs to other tenant and is not public. or we can only get the wafpolicy which is in our own tenant (with the {tenantId: tenantId} option parameter).

Comment (zongzw) [reply app/waf/src/controllers/wafpolicy.controller.ts]: ... Got it. Thanks.

**Pull request merged**